packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -9,98 +9,319 @@ import 'package:file/file.dart';
+  FlutterSkiaGoldFileComparator(
+    this.basedir, {
+      this.fs = const LocalFileSystem(),
+      this.platform = const LocalPlatform(),
+    }) : assert(basedir != null);
+
+  /// The directory to which golden file URIs will be resolved in [compare] and
+  /// [update].
+  final Uri basedir;
+
+  /// The file system used to perform file access.
+  @visibleForTesting
+  final FileSystem fs;
+
+  /// A wrapper for the [dart:io.Platform] API.
+  ///
+  /// This is only for use in tests, where the system platform (the default) can
+  /// be replaced by mock platform instance.
+  @visibleForTesting
+  final Platform platform;
+
+  final SkiaGoldClient _skiaClient = SkiaGoldClient();
+
+  /// Creates a new [FlutterSkiaGoldFileComparator] that mirrors the relative
+  /// path resolution of the default [goldenFileComparator].
+  ///
+  /// The [goldens] and [defaultComparator] parameters are visible for testing
+  /// purposes only.
+  static Future<FlutterSkiaGoldFileComparator> fromDefaultComparator({
+    SkiaGoldClient goldens,
+    LocalFileComparator defaultComparator,
+  }) async {
+    defaultComparator ??= goldenFileComparator;
+    goldens ??= SkiaGoldClient();
+
+    // Calculate the appropriate basedir for the current test context.
+    final FileSystem fs = goldens.fs;
+    final Directory testDirectory = fs.directory(defaultComparator.basedir);
+    final String testDirectoryRelativePath = fs.path.relative(testDirectory.path, from: goldens.flutterRoot.path);
+    return FlutterSkiaGoldFileComparator(goldens.comparisonRoot.childDirectory(testDirectoryRelativePath).uri);
+  }
+
+  /// Updates the local golden file with the most recent imageBytes before
+  /// uploading to Skia Gold for comparison.
+  ///
+  /// The returned future completes with a boolean value that indicates whether
+  /// [imageBytes] passed the Skia Gold pixel test.
+  ///
+  /// In the case of the given golden file being unavailable, the comparator
+  /// will throw a [TestFailure].
+  @override
+  Future<bool> compare(Uint8List imageBytes, Uri golden) async {
+    golden = _addPrefix(golden);
+    await update(golden, imageBytes);
+
+    final File goldenFile = getGoldenFile(golden);
+    if (!goldenFile.existsSync()) {
+      throw TestFailure('Could not be compared against non-existent file: "$golden"');
+    }
+    await _skiaClient.auth(fs.directory(basedir));
+    await _skiaClient.imgtestInit();
+    return await _skiaClient.imgtestAdd(golden.path, goldenFile);
+  }
+
+  /// Updates the golden file identified by [golden] with [imageBytes].
+  ///
+  /// This will be invoked in lieu of [compare] when [autoUpdateGoldenFiles]
+  /// is `true` (which gets set automatically by the test framework when the
+  /// user runs `flutter test --update-goldens`).
+  ///
+  /// The [FlutterSkiaGoldFileComparator] will always update the local golden
+  /// file so upload to Skia Gold.
   @override
   Future<void> update(Uri golden, Uint8List imageBytes) async {
```

Comment:
This impl is the same as in the other comparator. Should it live in the common base class?